weave/integrations/dspy/dspy_test.py

```diff
@@ -0,0 +1,147 @@
+import os
+from typing import Any
+
+import pytest
+
+import weave
+from weave.trace_server import trace_server_interface as tsi
+from weave.weave_client import WeaveClient
+
+
+def _get_call_output(call: tsi.CallSchema) -> Any:
+    """This is a hack and should not be needed. We should be able to auto-resolve this for the user.
+    Keeping this here for now, but it should be removed in the future once we have a better solution.
+    """
+    call_output = call.output
+    if isinstance(call_output, str) and call_output.startswith("weave://"):
+        return weave.ref(call_output).get()
+    return call_output
+
+
+@pytest.mark.skip_clickhouse_client
+@pytest.mark.vcr(
+    filter_headers=["authorization"],
+    allowed_hosts=["api.wandb.ai", "localhost", "trace.wandb.ai"],
+)
+def test_dspy_language_models(client: WeaveClient) -> None:
+    import dspy
+
+    os.environ["DSP_CACHEBOOL"] = "False"
+
+    gpt3_turbo = dspy.OpenAI(
+        model="gpt-3.5-turbo-1106",
+        max_tokens=300,
+        api_key=os.environ.get("OPENAI_API_KEY", "DUMMY_API_KEY"),
+    )
+    dspy.configure(lm=gpt3_turbo)
+    prediction = gpt3_turbo("hello! this is a raw prompt to GPT-3.5")
+    expected_prediction = "Hello! How can I assist you today?"
+    assert prediction == [expected_prediction]
+    weave_server_respose = client.server.calls_query(
+        tsi.CallsQueryReq(project_id=client._project_id())
+    )
+    assert len(weave_server_respose.calls) == 4
+
+    call_1 = weave_server_respose.calls[0]
```

Comment:
please assert the name and ordering of calls. See example from Langchain integration here: https://github.com/wandb/weave/blob/9cd38e8e88f2ea0d3f8822ef835e02672f4bcff0/weave/integrations/langchain/langchain_test.py#L35